src/core_plugins/kibana/public/dashboard/dashboard_strings.js

```diff
@@ -0,0 +1,48 @@
+import { DashboardViewMode } from './dashboard_view_mode';
+import _ from 'lodash';
+
+export class DashboardStrings {
+
+  /**
+   * @param list {Array.<string>}
+   * @returns {string} The list of strings concatenated with commas so it can be used in a message.
+   * E.g. ['a', 'b', 'c'] returns 'a, b, and c'.
+   */
+  static createStringList(list) {
+    const listClone = _.clone(list);
+    const isPlural = list.length > 1;
+    const lastEntry = isPlural ? `, and ${list[list.length - 1]}` : '';
+    if (isPlural) listClone.splice(-1, 1);
+
+    return `${listClone.join(', ')}${lastEntry}`;
+  }
+
+  /**
+   * @param changedFilters {Array.<string>} An optional list of filter types that have changed.
+   * @returns {string} A warning message to display to the user that they are going to lose changes.
+   */
+  static getUnsavedChangesWarningMessage(changedFilters) {
+    const changedFilterList = this.createStringList(changedFilters);
+
+    return changedFilterList ?
+      `Are you sure you want to cancel and lose changes, including changes made to your ${changedFilterList}?` :
+      `Are you sure you want to cancel and lose changes?`;
+  }
+
+  /**
+   * @param title {string} the current title of the dashboard
+   * @param viewMode {DashboardViewMode} the current mode. If in editing state, prepends 'Editing ' to the title.
+   * @param isDirty {boolean} if the dashboard is in a dirty state. If in dirty state, adds (unsaved) to the
+   * end of the title.
+   * @returns {string} A title to display to the user based on the above parameters.
+   */
+  static getDashboardTitle(title, viewMode, isDirty) {
+    const isEditMode = viewMode === DashboardViewMode.EDIT;
+    const unsavedSuffix = isEditMode && isDirty
+      ? ' (unsaved)'
+      : '';
+
+    const displayTitle = `${title}${unsavedSuffix}`;
+    return isEditMode ? 'Editing ' + displayTitle : displayTitle;
+  }
+}
```

Comment:
Personally I'd do what @cjcenizal describes: exporting them separately. 

I love that you included the reasoning. There's definitely pros and cons. Here's the reasoning behind my opinion:

- You still have a namespace (the file), so you can for example `import * as DashboardStrings from './file'` (this will also give you autocompletion)
- It gives you the opportunity of only importing the one function you need (for this specific use-case that's not necessarily relevant, but for other utils it might be). In general I really like being able to `import { utilFn } from './someUtilFile'`.
- I haven't seen this pattern using `class` + `static` in JavaScript before. Not necessarily a relevant point at all, just an observation.
- If you push this further to a lib perspective — so not really interesting in the Kibana perspective, but more for the higher-level reason for my opinion — doing this [enables tree-shaking](https://webpack.js.org/guides/tree-shaking/) (aka dead-code elimination). Again, not necessarily relevant for Kibana, just why this is perhaps a more used approach in the JS world.